### DISCUSSION TITLE
Validation doesn't work

### DISCUSSION THREAD

**Discussion by beqramo** [react-hook-form/react-hook-form#5502] [Q&A] [answered]
**Describe the bug**
Hi, I'm using this package for more than a year but now I have a problem with React native, in short validation doesn't work anymore starting from yesterday, and I don't understand what happened, I tried to downgrade but the same thing.  I'm doing super simple stuff, I checked on the Web and it worked. 

With `setError` it setting an error on the field, `trigger` and submit press does nothing. 

So I'm logging formState to see errors but it is empty

thank you in advance.

**To Reproduce**
press submit and it will not throw an error

**Codesandbox link (Required)**
sorry but I can't do it for RN, I will paste a little snippet here:

```tsx
// app.tsx // entry file
const App = () => {

  const { handleSubmit, control, formState, trigger, setError } = useForm({
    defaultValues: {
      email: '',
    },
  });
  console.log(formState, 'formState'); // errors object is always empty
  return (
    <Column >
      <Controller
        control={control}
        name="email"
        render={({ field: { onChange, onBlur, value } }) => (
          <TextInput
            onBlur={onBlur}
            onChangeText={(value) => onChange(value)}
            value={value}
            placeholder={'email'}
          />
        )}
        rules={{ required: true, minLength: 8 }}
      />
    </Column>

    <Button title="Submit" onPress={handleSubmit(() => {})} />)
}
```

**Expected behavior**
show error

**Screenshots**
this is a log on submit but indeed there should be the object of errors with data.
![image](https://user-images.githubusercontent.com/40486471/120905851-03f0ab00-c666-11eb-8b9b-22dd3f274899.png)

**Smartphone (please complete the following information):**

- Device:  iPhone12
- OS: 14.5
`"react-hook-form": "^7.8.1"`

**Comment by bluebill1049** [comment]
https://snack.expo.io/@bluebill1049/react-hook-form-v7---controller

**Reply by bluebill1049** [reply]
can you modify the link above and reproduce the problem?

**Reply by DracotMolver** [reply]
@bluebill1049  I'm facing the same problem and the snack expo does not validate. I used my device to try and when I press the button doesn't display the error messages, doesn't show the data when submit the button either.

**Comment by thanh-nguyen-95** [comment, marked as answer]
Have you tried passing `name` and `ref` props to the `TextInput`?

```jsx
<Controller
  control={control}
  name="email"
  render={({ field: { onChange, onBlur, value, name, ref /* here */ } }) => (
    <TextInput
      onBlur={onBlur}
      onChangeText={(value) => onChange(value)}
      value={value}
      placeholder={'email'}
      name={name}
      ref={ref}
    />
  )}
  rules={{ required: true, minLength: 8 }}
/>
```

If that doesn't work, please provide a CodeSandbox so we can take a closeer look at the code.

**Comment by beqramo** [comment]
Thanks guys, code was correct that I had, it was a native module error that was causing somehow this. in more specific `@storybook/react-native`, this error wasn't appearing in debug server and that is why it was hard to debug.

 thank you again and have a great day.

**Reply by DracotMolver** [reply]
Hi @beqramo  what issue did you have with `@storybook/react-native`? I'm asking because I'm facing the same situation. I have the latest version of RHF and I don't see the errores, neither work the `handleSubmit` method.

**Reply by beqramo** [reply]
To be honest I investigate little and I wasn't able to figure out what was the exact problem, maybe it is a proxy problem, I don't know, sorry
